src/extractor/intersection/intersection_analysis.cpp

```diff
@@ -706,6 +708,21 @@ IntersectionView convertToIntersectionView(const util::NodeBasedDynamicGraph &gr
         }
     }
 
+    auto no_uturn = std::none_of(
+        pre_intersection_view.begin(),
+        pre_intersection_view.end(),
+        [&is_uturn](IntersectionViewDataWithAngle road) { return is_uturn(road.first.angle); });
+    // After all of this, if we now don't have a u-turn, let's add one to the intersection.
+    // This is a hack to fix the triggered assertion ( see:
+    // https://github.com/Project-OSRM/osrm-backend/issues/6218 ). Ideally we would fix this more
+    // robustly, but this will require overhauling all of the intersection logic.
+    if (no_uturn)
+    {
+        BOOST_ASSERT(!uturn.entry_allowed && allowed_uturns_number > 0);
+        BOOST_ASSERT(uturn.angle == 0.);
```

Comment:
Same comment as above. The other assertion is because we only manually insert here if we didn't above.